### ISSUE TITLE
Able to create Azure/vSphere RKE node template without supplying cloud credential

### ISSUE THREAD

**Issue by davidnuzik** [rancher/rancher#18731]
Version: Master 2.2 (3/7/19) a3f8c48fe1e6

**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
- Create an Azure RKE cluster for the first time
- Add a node template, do NOT fill out required Cloud Credential fields, instead fill out just the node template name field near the bottom and click the "Create" button.
- Do this again for vSphere RKE
**UPDATE:**
`cloudCredentialId` is not marked as required in the schema `/v3/schemas/nodeTemplate`

**Result:**
You should not be allowed to Create a node template if we did not fill out Cloud Credentials details.

**Other details that may be helpful:**
When I do this my Azure rke cluster obviously fail to provision and cite missing subscription-id.
When I do this I also cannot ever delete my node templates when they are created this way.

Possibly related in some way to: #18093

**Comment by westlywright** [comment]
`cloudCredentialId` is not marked as required in the schema `/v3/schemas/nodeTemplate`, this should be moved to API issue. When it's marked as required the UI validation will pick it up and fire an error.

**Comment by alena1108** [comment]
@kinara I don't think we can make the field required via rancher/types as it would apply to **all** the node drivers' templates, and cloud credential support is not in for custom drivers yet. The fix can be on the api layer - just throw an error when cloud credential is not provided for the drivers we know support cloud creds (DO, Azure, etc)

**Comment by davidnuzik** [comment]
Version: Master 2.2 (3/11/19) 540d086476c9
Confirmed issue. I cannot create an Azure or vSphere node template unless I supply a Cloud Credential. If I attempt to do so, I get an appropriate error in the UI:
`Validation failed in API: Cloud Credential must be set is required`

API returns:
`{"baseType":"error","code":"MissingRequired","message":"Cloud Credential must be set","status":422,"type":"error"}`

I smoke tested this area and other providers to ensure nothing broke. I encountered no issues during my testing.
